src/server/storage/storageAPI.js

```diff
@@ -179,5 +181,54 @@ const setup = (app: Router, storage: StorageAPI) => {
       res.json({ ok: 1, result })
     })
   )
+
+  /**
+   * @api {get} /login/token get W3 login token for current user
+   * @apiName Get
+   * @apiGroup Storage
+   *
+   * @apiSuccess {Number} ok
+   * @apiSuccess {String} loginToken
+   * @ignore
+   */
+  app.get(
+    '/storage/login/token',
+    passport.authenticate('jwt', { session: false }),
+    wrapAsync(async (req, res, next) => {
+      const { user, log } = req
+      const logger = log.child({ from: 'storageAPI - login/token' })
+
+      let loginToken = user.loginToken
+
+      if (!loginToken) {
+        const secureHash = md5(user.email + conf.secure_key)
+        const web3Response = await fetch(`${conf.web3SiteUrl}/api/wl/user`, {
```

Comment:
Maybe transferred to middleware or to some helper?

Reply:
done